docs/development/i18n-guide.md

```diff
@@ -0,0 +1,261 @@
+# Renovate i18n Guide
+
+This document briefs how to provide translatable to user-facing text, how to contribute code for i18n, and what is your recommended way/style to do that.
+
+This document quotes some definitions and segments from the manual of [GNU Gettext](https://www.gnu.org/software/gettext/manual/gettext.html).
```

Comment:
Please keep this minimal:
- Use a link rather than copy/paste where possible,
- make sure it's clear when the text is a quotation